web/src/user_sort.ts

```diff
@@ -1,13 +1,18 @@
-export function compare_a_b(a, b) {
-    if (a > b) {
+import type {User} from "./people";
+
+export function compare_a_b<T extends keyof User>(a: User[T], b: User[T]): number {
+    // Setting default value to 0 if a or b is null or undefined
+    const aValue = a !== null && a !== undefined ? a : 0;
+    const bValue = b !== null && b !== undefined ? b : 0;
```

Comment:
I think this function expects `string` or `integer`, and does not ever expect `null`. I'd prefer to not change it to try to handle `null`. `sort_email` clearly excludes the possibility of `null` being passed here.

I also suspect `people.ts` has a bug in the type for `delivery_email`; the server code is pretty clear that it can be `null`, and never `undefined`.

Reply:
Although null checks are performed before calling the function, TypeScript still issues warnings about null values. Therefore, assertions are added to the variables to address these warnings. 


About `people.ts` because the property `delivery_email` is optional in `people.ts` so it's value can be undefined here. I looked at the server code I don't think `delivery_email` could be optional there.